src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2246,10 +2246,40 @@ theorem toNat_udiv {x y : BitVec n} : (x / y).toNat = x.toNat / y.toNat := by
   · rw [toNat_ofNat, Nat.mod_eq_of_lt]
     exact Nat.lt_of_le_of_lt (Nat.div_le_self ..) (by omega)
 
+@[simp]
+theorem zero_udiv {x : BitVec w} : (0#w) / x = 0#w := by
+  simp [bv_toNat]
+
 @[simp]
 theorem udiv_zero {x : BitVec n} : x / 0#n = 0#n := by
   simp [udiv_def]
 
+@[simp]
+theorem udiv_one {x : BitVec w} : x / 1#w = x := by
+  simp only [udiv_eq, toNat_eq, toNat_udiv, toNat_ofNat]
+  cases w
+  · simp [eq_nil x]
+  · simp
+
+@[simp]
+theorem udiv_eq_and {x y : BitVec 1} :
+    x / y = (x &&& y) := by
+  have hx : x = 0#1 ∨ x = 1#1 := by bv_omega
+  have hy : y = 0#1 ∨ y = 1#1 := by bv_omega
+  rcases hx with rfl | rfl <;>
+    rcases hy with rfl | rfl <;>
+      rfl
```

Comment:
```suggestion
    x / y = (x &&& y) := by revert x y; decide
```

Since this statement has only finitely many cases to check, we can use the `Decidable` instances to perform this checking for us!